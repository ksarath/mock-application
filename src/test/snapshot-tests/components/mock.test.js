import React from 'react';
import renderer from 'react-test-renderer';
import Mock from '../../../main/js/components/mock';

test('Mock component changes the class when hovered', () => {
  const component = renderer.create(
    <Mock />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.children[0].props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.children[0].props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
