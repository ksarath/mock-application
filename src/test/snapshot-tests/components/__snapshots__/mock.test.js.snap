// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mock component changes the class when hovered 1`] = `
<div
  className="normal"
>
  <img
    alt="Mock Application Image"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    src="test-file-stub"
  />
</div>
`;

exports[`Mock component changes the class when hovered 2`] = `
<div
  className="hovered"
>
  <img
    alt="Mock Application Image"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    src="test-file-stub"
  />
</div>
`;

exports[`Mock component changes the class when hovered 3`] = `
<div
  className="normal"
>
  <img
    alt="Mock Application Image"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    src="test-file-stub"
  />
</div>
`;
